sdk/core/System.ClientModel/src/Internal/TrimmingAttribute.cs

```diff
@@ -0,0 +1,420 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+namespace System.Diagnostics.CodeAnalysis;
```

Comment:
Why is this file needed?

Reply:
For the AOT decorators we use https://github.com/Azure/azure-sdk-for-net/pull/39478/files#diff-a3e46e71409e56111ac3637b674d02b46e7f7f7bbfbc60d35ebf4d2d268908e7R101